### PR TITLE
[Improve] Support JSON multi-level nested parsing

### LINKED ISSUE #9973: [Feature]Hope to support JSON multi-level nested parsing!
### Search before asking

- [x] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In Kafka, when parsing JSON strings, for example:
```shell
{“personInfo”:{“name”:“kk”,“age”:12}, “other”:12}
{ “other”:12}
```
It can be seen that the first line has the "personInfo" field, while the second line does not. However, the schema definition of the Kafka source is as follows:
```shell
personInfo = {
  name = string,
  age = int
}
other = int
```
The above definition is fine. But when performing SQL transformation in the transform step, the second line will cause a problem. For example, if the query is:
```shell
transform {
    Sql {
            plugin_input = "kafka_table"
            plugin_output = "out_table"
            query = """
              select personInfo.name as personInfo_name from kafka_table
            """
        }
}
```
The above code will throw an exception. The exception is as follows:
```shell
Caused by: org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException: org.apache.seatunnel.transform.exception.TransformException: ErrorCode:[TRANSFORM_COMMON-06], ErrorDescription:[The expression 'personInfo.name' of SQL transform execute failed]
        at org.apache.seatunnel.transform.exception.TransformCommonError.sqlExpressionError(TransformCommonError.java:79)
```
The specific reason is that "personInfo" is null in the second line of data. In this scenario, an exception should not be thrown; instead, null should be used for filling. It is hoped that SeaTunnel can add this nested parsing functionality.

---

在kafka中，解析json字符串，比如：
```shell
{“personInfo”:{"name":"kk","age":12}, "other":12}
{ "other":12}
```
可以看出第1行有personInfo这个字段，第2行没有。但是在kafka的source的schema定义如下：
```shell
personInfo = {
  name = string,
  age = int
}
other = int
```
上面的定义没有什么问题。但是在transform的Sql转换的时候第二行却会出现问题， 比如query为
```shell 
transform {
    Sql {
            plugin_input = "kafka_table"
            plugin_output = "out_table"
            query = """
              select personInfo.name as personInfo_name from kafka_table
            """
        }
}
```
上面的代码会抛出异常。异常如下：
```shell 
Caused by: org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException: org.apache.seatunnel.transform.exception.TransformException: ErrorCode:[TRANSFORM_COMMON-06], ErrorDescription:[The expression 'personInfo.name' of SQL transform execute failed]
        at org.apache.seatunnel.transform.exception.TransformCommonError.sqlExpressionError(TransformCommonError.java:79)
```
具体的原因是第二行数据中，personInfo 是 null， 在这个场景中，其实不应该抛出异常，而应该使用null进行填充即可。希望seatunnel能够加上这个嵌套解析的功能。

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by misi1987107**
### Purpose of this pull request

resolve https://github.com/apache/seatunnel/issues/9973

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)

### CODE REVIEW

**Review by davidzollo — DISMISSED** [review]
+1 if CI passed

**Review by zhangshenghang — APPROVED** [review]

**Review by zhangshenghang — CHANGES_REQUESTED** [review]
image

thanks ， Please merge dev and resolve the current CI issue

**Comment by misi1987107** [comment]
> +1 if CI passed

Thanks

**Comment by misi1987107** [comment]
> image
> thanks ， Please merge dev and resolve the current CI issue

Ok,I will do this

File: seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/sql/zeta/CastFunctionTest.java [hunk 1]
```diff
@@ -54,4 +55,29 @@ public void testCastFunction() {
...
+
+        SeaTunnelRowType outRowType = sqlEngine.typeMapping(null);
+
+        SeaTunnelRow outRow = sqlEngine.transformBySQL(inputRow, outRowType).get(0);
+
+        Object addressField = outRow.getField(0);
+        Assertions.assertNull(
+                addressField,
+                "When casting nested field where intermediate value is null, result should be null");
+    }
```
Comment (zhangshenghang) [inline seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/sql/zeta/CastFunctionTest.java:82]: Can you add some normal scenarios to ensure there are no other issues?

Comment (misi1987107) [reply seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/sql/zeta/CastFunctionTest.java:82]: OK，I will add some normal scenarios test cases.

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java [hunk 2]
```diff
@@ -340,6 +340,9 @@ public Object computeForValue(Expression expression, Object[] inputFields) {
                     }
                     parDataType = ((SeaTunnelRowType) parDataType).getFieldType(idx);
                     res = parRowValues.getFields()[idx];
+                    if (res == null) {
```
Comment (mosence) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java:343]: It seems that the MapType type is not supported data like this:

select a.b.c from t;

t table data:
```json
[{
  "a": {  // MapType
     "b": {
         "c": "a.b.c's Value"
     }
  }
},{
  "a": {  // MapType
     "b": {
     }
  }
},
{
   //... More RowData...
}]
```

**Review by davidzollo — DISMISSED** [review]
+1 if CI passed

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java (hunk 2 above)
Comment (mosence) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java:343]: If Json do not use this type: _MapType_. Ignore it.

Comment (dybyte) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java:343]: I believe this feature was discussed in #9747. I suggest limiting this PR to null handling only.  cc @zhangshenghang @davidzollo

**Comment by zhangshenghang** [comment]
thanks @misi1987107  
Please add a test here ，to verify if returning null can run successfully. https://github.com/apache/seatunnel/blob/dev/seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/java/org/apache/seatunnel/e2e/transform/TestSQLIT.java

**Review by zhangshenghang — APPROVED** [review]
LGTM

**Review by davidzollo — APPROVED** [review]
+1
